ports/nrf/modules/machine/uart.c

```diff
@@ -446,17 +456,30 @@ static mp_uint_t mp_machine_uart_write(mp_obj_t self_in, const void *buf, mp_uin
     #endif
 
     machine_uart_obj_t *self = self_in;
-    nrfx_err_t err = nrfx_uart_tx(self->p_uart, buf, size);
-    if (err == NRFX_SUCCESS) {
+
+    // Send data out, in chunks if needed.
+    mp_uint_t remaining = size;
+    while (remaining) {
+        #if defined(NRF52832)
+        // The nRF52832 cannot write more than 255 bytes at a time.
+        mp_uint_t chunk = MIN(255, remaining);
+        #else
```

Comment:
Would it make sense to have this 255 number configurable?

Reply:
I've now made this a macro.